src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs

```diff
@@ -111,8 +111,8 @@
 
 const string SimpleVecOpTest_ValidationLogicForCndSelForNarrowing = @"for (var i = 0; i < Op1ElementCount; i++)
             {
-                {Op1BaseType} iterResult = (mask[i] != 0) ? {GetIterResult} : falseVal[i];
-                if (iterResult != result[i])
+                {RetBaseType} iterResult = (mask[i] != 0) ? {GetIterResult} : falseVal[i];
+                if ({ConvertFunc}(iterResult) != {ConvertFunc}(result[i]))
```

Comment:
can you please run the tests that uses the templates updated to make sure they pass?

Reply:
Sure: The only template using this validation logic right now is `SveSimpleVecOpDifferentRetTypeTest`, which is only used by the `ConvertTo*` APIs (for now) and `FloatingPointExponentialAccelerator`. Both are passing.